packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -717,27 +728,27 @@ function ResizableTableColumnHeader(props) {
         layout.onColumnResizeStart(column.key);
         setIsInResizeMode(true);
         break;
+      default:
+        if (menuProps.onAction) {
+          menuProps.onAction(key);
```

Comment:
Shouldn't we also pass the column key to onAction here (would need to figure out how to handle the typing though...)? I feel like there isn't actually enough information provided to the user for them to actually perform a custom action like `hide` (their onAction would receive `hide` but they wouldn't know it was triggered on column X)

Reply:
I intentionally left it off, 1) people can set it on the handler itself, there should be some examples in the stories I added that handle this 2) onAction is provided on the Menu, and Menu has no idea about columns, I could add/inject it, but then, as you said, the type returned would be wildly different

Reply:
here's an example https://github.com/adobe/react-spectrum/pull/3886/files#diff-54aeb21d2962e015152a7810fb62771f0fc10bfadaa32da552c66f21be2a6aa9R1634

Reply:
thanks for the example link. IMO it just felt weird that we make the user do all that extra work when we already know what column key is associated with the menu action in `onMenuSelect` in `ResizableTableColumnHeader`. Thinking out loud, maybe we could provide a Table column specific Menu component that wraps Menu but supports a modified onAction type. Perhaps a bit too much

Reply:
lets continue the conversation up here https://github.com/adobe/react-spectrum/pull/3886#discussion_r1065203326